CharlieBackend.Business/Services/CourseService.cs

```diff
@@ -20,76 +20,55 @@ public CourseService(IUnitOfWork unitOfWork, IMapper mapper)
             _mapper = mapper;
         }
 
-        public async Task<Result<CourseDto>> CreateCourseAsync(CreateCourseDto courseDto)
+        public async Task<Result<CourseDto>> CreateCourseAsync(CreateCourseDto courseModel)
         {
             try
             {
-                if (courseDto == null)
-                {
-                    return Result<CourseDto>.GetError(ErrorCode.ValidationError, "CourseDto is null");
-                }
-
-                if (await IsCourseNameTakenAsync(courseDto.Name))
-                {
-                    return Result<CourseDto>.GetError(ErrorCode.UnprocessableEntity, "Course already exists");
-                }
-
-                var createdCourseEntity = _mapper.Map<Course>(courseDto);
+                var createdCourseEntity = _mapper.Map<Course>(courseModel);
 
                 _unitOfWork.CourseRepository.Add(createdCourseEntity);
 
                 await _unitOfWork.CommitAsync();
+                var createdCourseDto = _mapper.Map<CourseDto>(createdCourseEntity);
 
-                return Result<CourseDto>.GetSuccess(_mapper.Map<CourseDto>(createdCourseEntity));
+                return Result<CourseDto>.GetSuccess(createdCourseDto);
             }
-            catch
+            catch 
             {
+
                 _unitOfWork.Rollback();
 
-                return Result<CourseDto>.GetError(ErrorCode.InternalServerError, "Internal error");
+                return null;
```

Comment:
Why you pick up null?

Reply:
that was after resolving conflicts
maybe smth went wrong